.travis.yml

```diff
@@ -1,47 +1,72 @@
-
+sudo: false
+os:
+  - linux
+  - osx
 language: cpp
-
 compiler:
  - clang
 
 env:
- - GOPATH=${TRAVIS_BUILD_DIR}/go CPPFLAGS="-I/usr/src/gmock/include" CXXFLAGS="-Wall -Werror"
+  - CXXFLAGS="-std=c++11"
 
-cache: apt
+cache:
+  apt: true
+  directories:
+    - $HOME/.cache/pip
+    - $TRAVIS_BUILD_DIR/../ct
+    - $TRAVIS_BUILD_DIR/../depot_tools
+    - $TRAVIS_BUILD_DIR/../install
+
+addons:
+  apt:
+    sources:
+      - ubuntu-toolchain-r-test
+      - llvm-toolchain-precise-3.6
+    packages:
+    - autoconf
+    - automake
+    - build-essential
+    - clang-3.6
+    - python-dev
+    - tcl8.5
+    - libstdc++-4.9-dev
 
 before_install:
- - sudo apt-add-repository -y ppa:sao/backports
- - sudo apt-add-repository -y ppa:ubuntu-toolchain-r/test
- - sudo apt-add-repository -y ppa:fcitx-team/nightly
- - sudo apt-add-repository -y ppa:chris-lea/protobuf
- - sudo apt-add-repository -y ppa:asolovets/backports
- - sudo apt-get update -qq
- - sudo apt-get install -qq openssl libssl-dev autoconf automake protobuf-compiler libprotobuf-java libprotobuf-dev python-dev libjson-c-dev libgoogle-glog-dev libgflags-dev libldns-dev libstdc++-4.8-dev libleveldb-dev libsnappy-dev libgoogle-perftools-dev
-# Stupid frikkin' google-mock package on Precise is b0rked, so hack it up:
- - wget https://googlemock.googlecode.com/files/gmock-1.7.0.zip -O /tmp/gmock-1.7.0.zip
- - unzip -d /tmp /tmp/gmock-1.7.0.zip
- - sudo mv /tmp/gmock-1.7.0 /usr/src/gmock
- - wget https://github.com/ellzey/libevhtp/archive/1.2.10.zip -O /tmp/libevhtp-1.2.10.zip
- - unzip -d /tmp/ /tmp/libevhtp-1.2.10.zip
- - mkdir /tmp/libevhtp-1.2.10/build
- - pushd /tmp/libevhtp-1.2.10/build && cmake -DEVHTP_DISABLE_REGEX:STRING=ON -DCMAKE_C_FLAGS="-fPIC" .. && sudo make install; popd
- - sudo pip install -r python/requirements.txt
- - sudo ln -s protobuf-java.jar /usr/share/java/protobuf.jar
+ - if [ "$CXX" == "clang++" ] && which clang++-3.6; then export CXX="clang++-3.6" CC="clang-3.6"; fi
+ - if [ "$CXX" == "g++" ] && which g++-4.8; then export CXX="g++-4.8" CC="gcc-4.8"; fi
+ - if [ "$TRAVIS_OS_NAME" == "osx" ]; then . ./.travis.osx.before_install.sh; fi
+ - export INSTALL_DIR=$TRAVIS_BUILD_DIR/../install
+ - mkdir -p ${INSTALL_DIR}/include ${INSTALL_DIR}/lib ${INSTALL_DIR}/java
+ - echo "INSTALL_DIR=${INSTALL_DIR}"
+ - export PROTOC="${INSTALL_DIR}/bin/protoc"
+ - export GOPATH=${INSTALL_DIR}/go
  - mkdir -p $GOPATH/src/github.com/google
- - ln -s $PWD $GOPATH/src/github.com/google
+ - ln -sf $PWD $GOPATH/src/github.com/google
+ - echo "jars.protobuf = ${INSTALL_DIR}/java/protobuf.jar" > local.properties
+ - echo "protoc.bin = ${PROTOC}" >> local.properties
+
+install:
+ - autoconf --version
+ - pushd ..
```

Comment:
They have a specific note that says not to depend on `HOME`, so maybe this will have to do... It should break loudly if it stops working, anyway. Or we could change to subdirs, but no rush.